<launch>

  <arg name="marker_size" default="5.6" />
  <arg name="max_new_marker_error" default="0.08" />
  <arg name="max_track_error" default="0.2" />
  <arg name="cam_image_topic" default="/usb_cam/image_raw" />
  <arg name="cam_info_topic" default="/usb_cam/camera_info" />
  <arg name="output_frame" default="/usb_cam" />
  <arg name="mname" default="zumy6" />

  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node"  output="screen" >
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="1280" />
    <param name="image_height" value="720" />
    <param name="pixel_format" value="mjpeg" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap" />
    <param name="camera_info_url" value="file:////home/cc/ee106a/fa15/class/ee106a-bd/ros_workspaces/lab6/lifecam.yml" />
  </node>

  <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen" args="$(arg marker_size) $(arg max_new_marker_error) $(arg max_track_error) $(arg cam_image_topic) $(arg cam_info_topic) $(arg output_frame)" />

  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/usb_cam/image_raw"/>
    <param name="autosize" value="true" />
  </node>

  <node name="image_view_cv" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/processed_image"/>
    <param name="autosize" value="true" />
  </node>

<!--   <node name="object_tracker" pkg="lab6" type="object_tracker.py" respawn="false" output="screen" /> -->

  <node name="object_tracker_red" pkg="lab6" type="object_tracker_red.py" respawn="false" output="screen" />
  <node name="object_tracker_gbo" pkg="lab6" type="object_tracker_gbo.py" respawn="false" output="screen" />

<!--   <node name="object_tracker_blue" pkg="lab6" type="object_tracker_blue.py" respawn="false" output="screen" />
  <node name="object_tracker_green" pkg="lab6" type="object_tracker_green.py" respawn="false" output="screen" />
  <node name="object_tracker_orange" pkg="lab6" type="object_tracker_orange.py" respawn="false" output="screen" /> -->

<!--
  <include file="$(find odroid_machine)/launch/odroid_machine.launch">
    <arg name="mname" value="$(arg mname)" />
  </include>

  <node machine="$(arg mname)" ns="$(arg mname)" name="zumy_ros_bridge" pkg="ros_zumy" type="zumy_ros_bridge.py"  output="screen" >
  </node>
-->
</launch>

